File: Python/C_Strings/Exercicio_1277.py
leia = int(input())
for x in range(leia):
    numeroEstudantes = int(input())
    estudantes = input()
    frequencia = input()
    estudantes = estudantes.split()
    frequencia = frequencia.split()
    faltosos = ""
    for z in range(numeroEstudantes):
        falta = 0
        presença = 0
        porcentagem = 0
        aluno = estudantes[z]
        numeroFreq = frequencia[z]
        total = len(numeroFreq)
        for p in range(len(numeroFreq)):
            
            if frequencia[z][p] =="P":
                presença = presença + 1
            elif frequencia[z][p]== "M":
                total = total - 1
        porcentagem = (presença/total)*100
        total = 0
        if porcentagem<75:
            faltosos = faltosos + " " +aluno
    print(faltosos.strip())

File: Javascript/C_Strings/Exercicio_1277.js
var input = require("./readFile.js").input;
var lines = input.split('\n');
var a = 1
for(let i = 1;i<=parseInt(lines[0]);i++){
    var n = lines[a]
    var vagabundos = ""
    a+=1
    var estudantes = lines[a].split(" ")
    a+=1
    var faltas = lines[a].trim().split(" ")
    a+=1
    var e = 0
    for(let falta of faltas){
    var total = falta.length
    var presencas = falta.match(RegExp("P", 'g')) || 0
    var atestados = falta.match(RegExp("M", 'g')) || 0
    var porcentagem = (presencas.length/(total - (atestados.length ||0))*100) || 0
    if(porcentagem<75){
    vagabundos+= estudantes[e] + " "
    }
    e+=1
    }
    console.log(vagabundos.trim())
}